### ISSUE TITLE
Output spam from plugin when syncing into place built by Rojo

### ISSUE THREAD

**Issue by ayuka** [rojo-rbx/rojo#465]
I've fully migrated our project from a partially-managed setup to a fully-managed one now, and we're using the latest Rojo 7 alpha build due to the support of the binary format. Thank you for that!

However, when live-syncing into a place built by this it causes a huge lag spike and then major output spam. I think it spams well over 10,000 times but the studio output can only contain 5,000 at one time.

This is a large project and I'm unable to send a full build that reproduces this, however I am fully willing to go through any amount of steps necessary to help resolve this issue. I will try to create a sharable place file that reproduces this.

`[Rojo-Warn] Failed to decode property of type Ref (x4994)`

**Comment by kennethloeffler** [comment]
We recently landed a fix related to `Ref` sync in #462. If you have cargo installed, you can try `cargo install --git https://github.com/rojo-rbx/rojo` to build Rojo off the master branch. It may or may not fix your issue but it's worth a shot!

**Comment by ayuka** [comment]
I just installed from the repository and built the latest plugin from the master branch. I'm now receiving another message which is more obscure (?): `[Rojo-Warn] Failed to decode property of type nil (x4995)`

Since I have the source now, I'll see if I can debug a bit with it.

**Comment by kennethloeffler** [comment]
Make sure to also reinstall Rojo's plugin using `rojo plugin install`

**Comment by ayuka** [comment]
Yes, I built the plugin from source instead from the master branch and tried using the CLI install.

I printed off what was being passed and I'm getting this:
```
  11:26:26.428   ▼  {
                    ["Ref"] = "00000000000000000000000000000000"
                 }  -  Edit
  11:26:26.429  [Rojo-Warn] Failed to decode property of type nil  -  Edit
```

**Comment by kennethloeffler** [comment]
It seems like you're still running the old plugin (I've experienced the same problem while working on it 😄). In previous versions of Rojo's protocol, property values had the fields `Type` and `Value`, but now they're of the form you see in the print. I think the plugin is still trying to access the `Type` field (which doesn't exist anymore) and is getting `nil`, hence the warning and failure to sync.

Uninstall or disable any other instances of the Rojo plugin in Roblox Studio, run `rojo plugin install`, then open a new place (or restart Studio) to make sure that the correct plugin is running.

**Comment by ayuka** [comment]
I've managed to create a simple reproduction of this issue. It mostly seems that all object properties when nil cause this issue, on top of the old implementation from the start.

Regarding the version of the plugin, the master branch as of now has the line of code that I'm having trouble with: https://github.com/rojo-rbx/rojo/blob/master/plugin/src/Reconciler/diff.lua#L78

Reproduction file: [refbug.zip](https://github.com/rojo-rbx/rojo/files/6956057/refbug.zip)

**Comment by ayuka** [comment]
I've seemingly resolved the issue, I'll create a pull request shortly that hopefully resolves this.

Thank you for your help!

Also, after some more looking, it seems that properties that have `inf` as a value pass an empty table. I'm not too well-versed in the Rojo protocol, so I can't help with this specifically. If you'd like, I can create a new issue for this specifically. Since the Type is no longer passed, it may be a good idea to change how this specific warning is shown to prevent potentially vague messages going forward.

```
  12:21:13.359  MotorMaxAcceleration Ragdoll_LeftLowerArm  -  Edit
  12:21:13.359  [Rojo-Warn] Failed to decode property of type nil  -  Edit
  12:21:13.359  MotorMaxAcceleration Ragdoll_RightLowerArm  -  Edit
  12:21:13.360  [Rojo-Warn] Failed to decode property of type nil  -  Edit
  12:21:13.360  MotorMaxAcceleration Ragdoll_LeftLowerLeg  -  Edit
  12:21:13.360  [Rojo-Warn] Failed to decode property of type nil  -  Edit
  12:21:13.360  MotorMaxAcceleration Ragdoll_RightLowerLeg  -  Edit
  12:21:13.360  [Rojo-Warn] Failed to decode property of type nil  -  Edit
  12:21:13.360  MotorMaxAcceleration HeadCollision  -  Edit
  12:21:13.360  [Rojo-Warn] Failed to decode property of type nil  -  Edit
  12:21:15.661  MaxDistance NameDisplay  -  Edit
  12:21:15.661  [Rojo-Warn] Failed to decode property of type nil  -  Edit
  12:21:15.661  MaxDistance BillboardGui  -  Edit
  12:21:15.661  [Rojo-Warn] Failed to decode property of type nil  -  Edit
  12:21:15.673  MaxAngularVelocity AlignOrientation  -  Edit
  12:21:15.673  [Rojo-Warn] Failed to decode property of type nil  -  Edit
  12:21:15.714  MaxDistance NPCQuestIndicator  -  Edit
  12:21:15.714  [Rojo-Warn] Failed to decode property of type nil  -  Edit
  12:21:15.729  MaxDistance NPCQuestIndicator  -  Edit
  12:21:15.729  [Rojo-Warn] Failed to decode property of type nil  -  Edit
  12:21:15.740  MaxDistance NPCQuestIndicator  -  Edit
  12:21:15.740  [Rojo-Warn] Failed to decode property of type nil  -  Edit
```

**Comment by kennethloeffler** [comment]
Oh, that's funny. Please do create another issue!
